tests/cocotb/test_stream_tcdm.py

```diff
@@ -0,0 +1,341 @@
+# ---------------------------------
+# Copyright 2024 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+#
+# Description:
+# This tests the basic configuration of the streamer
+# found in ./util/cfg/streamer_cfg.hjson
+#
+# Sequence of tests:
+# 1. First test read and write to CSR registers
+# 2. Check if the output addresses of TCDM
+#    are correct and valid
+# ---------------------------------
+
+import cocotb
+from cocotb.triggers import RisingEdge, Timer
+from cocotb.clock import Clock
+from cocotb_test.simulator import run
+import snax_util
+import os
+import subprocess
+from decimal import Decimal
+
+# Configurable design time parameters
+TCDM_REQ_PORTS = 12
+NARROW_DATA_WIDTH = 64
+WIDE_DATA_WIDTH = 512
+TCDM_DEPTH = 64
+NR_BANKS = 32
+NUM_INPUT = 2
+
+# Configurable testing parameters
+# In the default value below, the number
+# of tests fills the entire memory
+BANK_INCREMENT = int(NARROW_DATA_WIDTH / 8)
+WIDE_BANK_INCREMENT = int(WIDE_DATA_WIDTH / 8)
+WIDE_NARROW_RATIO = int(WIDE_DATA_WIDTH / NARROW_DATA_WIDTH)
+NUM_NARROW_TESTS = TCDM_DEPTH * NR_BANKS
+NUM_WIDE_TESTS = int(NUM_NARROW_TESTS / 8)
+MIN_VAL = 0
+MAX_NARROW_VAL = 2**NARROW_DATA_WIDTH
+MAX_WIDE_VAL = 2**WIDE_DATA_WIDTH
+
+# DON'T TOUCH ME PLEASE
+# CSR parameters from the default
+# Configuration found under util/cfg/streamer_cfg.hjson
+# Also some pre-computed that are fixed
+CSR_LOOP_COUNT_0 = 0
+CSR_TEMPORAL_STRIDE_0 = 1
+CSR_TEMPORAL_STRIDE_1 = 2
+CSR_TEMPORAL_STRIDE_2 = 3
+CSR_SPATIAL_STRIDE_0 = 4
+CSR_SPATIAL_STRIDE_1 = 5
+CSR_SPATIAL_STRIDE_2 = 6
+CSR_BASE_PTR_0 = 7
+CSR_BASE_PTR_1 = 8
+CSR_BASE_PTR_2 = 9
+CSR_START_STREAMER = 10
```

Comment:
Does this duplicated a bit as in the `test_basic_streamer.py`?

Reply:
It is I'm thinking of porting outside next time. Hmm maybe in the next PR 😄